From 0ecf7243553f9f2544293c130778eed3f2d85bb7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Nikola=20Forr=C3=B3?= <nforro@redhat.com>
Date: Tue, 8 Nov 2016 11:40:22 +0100
Subject: [PATCH] Update cat file timestamps to prevent using outdated version

---
 src/man.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/man.c b/src/man.c
index d4da309..475115a 100644
--- a/src/man.c
+++ b/src/man.c
@@ -30,6 +30,7 @@
 #include <errno.h>
 #include <unistd.h>
 #include <locale.h>
+#include <fcntl.h>
 #ifdef TERMIOS_HEADER
 #include <sys/termios.h>
 #endif
@@ -817,6 +818,7 @@ make_cat_file (const char *path, const char *man_file, const char *cat_file) {
      char *roff_command;
      char *command = NULL;
      struct stat statbuf;
+     struct timespec times[2];
 
      /* _Before_ first, make sure we will write to a regular file. */
      if (stat(cat_file, &statbuf) == 0) {
@@ -906,6 +908,14 @@ make_cat_file (const char *path, const char *man_file, const char *cat_file) {
 	  /* be silent about the success of chmod - it is not important */
 	  if (debug)
 	       gripe (CHANGED_MODE, cat_file, mode);
+     /* update cat file timestamps */
+     times[0].tv_sec = 0;
+     times[0].tv_nsec = UTIME_NOW;
+     times[1].tv_sec = 0;
+     times[1].tv_nsec = 0;
+     if (stat(man_file, &statbuf) == 0)
+          times[1] = statbuf.st_mtim;
+     utimensat(AT_FDCWD, cat_file, times, 0);
      } else {
 	  /* something went wrong - remove garbage */
 	  if(unlink(cat_file) != 0 && suid) {
-- 
2.7.4

