From 7d3dec11a86d2e367e59483d6414f3b53683b717 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Nikola=20Forr=C3=B3?= <nforro@redhat.com>
Date: Thu, 3 Nov 2016 11:00:28 +0100
Subject: [PATCH] Fix typo in man.1 man page

---
 man/en/man.man | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/man/en/man.man b/man/en/man.man
index 88fb6b1..6ec51de 100644
--- a/man/en/man.man
+++ b/man/en/man.man
@@ -91,7 +91,7 @@ Games et. Al.
 Miscellanea
 .TP
 .B 8
-System Administration tools and Deamons
+System Administration tools and Daemons
 .TP
 Distributions customize the manual section to their specifics, which often include additional sections.  
 
-- 
2.7.4

