diff -up man-1.6f/man/ja/man.man.jap man-1.6f/man/ja/man.man
--- man-1.6f/man/ja/man.man.jap	2005-08-21 01:26:06.000000000 +0200
+++ man-1.6f/man/ja/man.man	2012-01-20 14:55:38.219364683 +0100
@@ -66,9 +66,8 @@ 表示するプログラムを好きに選べる。
 MANPATH 中の空の文字列はデフォルトのリストを意味する。
 .TP
 .B \-\^P " pager"
-使用するページャーを指定する。デフォルトでは man は
-.BR @pager@ 
-を使う。このオプションは環境変数
+使用するページャーを指定する。
+このオプションは環境変数
 .B MANPAGER
 より優先する。この環境変数は
 .B PAGER
